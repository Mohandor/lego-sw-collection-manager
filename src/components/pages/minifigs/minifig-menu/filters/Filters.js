import React from "react";
import PropTypes from "prop-types";
import { Grid, TextField, MenuItem } from "@material-ui/core";
import styles from "./Filters.module.css";
import SelectField from "../../../../commons/inputs/selectfield";

export const filters = props => {
  const {
    showOptions,
    show,
    setShow,
    characNames,
    characNameSelected,
    setCharacName,
    resetCharacName,
    tags,
    tagSelected,
    setTag,
    resetTag
  } = props;

  const setShowHandler = e => {
    const value = e.target.value;
    value !== show && setShow(value);
  };
  return (
    <Grid container>
      <TextField
        id="select-show-filter"
        label="Show"
        select
        value={show}
        variant="outlined"
        fullWidth
        className={styles.Selectfield}
        onChange={setShowHandler}
      >
        {showOptions.map(option => (
          <MenuItem key={`${option}-selectfield-choice`} value={option}>
            {`${option[0].toUpperCase()}${option.slice(1)}`}
          </MenuItem>
        ))}
      </TextField>
      {characNames && (
        <SelectField
          value={characNameSelected}
          label="Character Name"
          options={characNames.map(charac => ({
            displayed: `${charac.name} (${charac.amount})`,
            value: charac.name
          }))}
          setValue={setCharacName}
          resetValue={resetCharacName}
        />
      )}
      {tags && (
        <SelectField
          value={tagSelected}
          label="Tag"
          options={tags.map(tag => ({
            displayed: `${tag.name} (${tag.amount})`,
            value: tag.name
          }))}
          setValue={setTag}
          resetValue={resetTag}
        />
      )}
    </Grid>
  );
};

filters.propTypes = {
  showOptions: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  setShow: PropTypes.func.isRequired,
  show: PropTypes.string.isRequired,
  tagSelected: PropTypes.string,
  tags: PropTypes.array,
  setTag: PropTypes.func.isRequired,
  resetTag: PropTypes.func.isRequired,
  characNameSelected: PropTypes.string,
  characNames: PropTypes.array,
  setCharacName: PropTypes.func.isRequired,
  resetCharacName: PropTypes.func.isRequired
};

export default filters;
