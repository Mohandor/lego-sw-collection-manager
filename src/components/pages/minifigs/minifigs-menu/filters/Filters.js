import React from "react";
import PropTypes from "prop-types";
import SelectField from "../../../../commons/inputs/selectfield";
import Inputs from "../../../../commons/inputs";

export const filters = props => {
  const {
    showOptions,
    show,
    setShow,
    characNames,
    characNameSelected,
    setCharacName,
    resetCharacName,
    tags,
    tagSelected,
    setTag,
    resetTag
  } = props;

  return (
    <>
      <Inputs
        type="radiobuttons"
        label="Show"
        value={show}
        updateInput={setShow}
        config={{
          row: true,
          options: showOptions
        }}
      />
      {characNames && (
        <SelectField
          value={characNameSelected}
          label="Character Name"
          options={characNames.map(charac => ({
            displayed: `${charac.name} (${charac.amount})`,
            value: charac.name
          }))}
          setValue={setCharacName}
          resetValue={resetCharacName}
        />
      )}
      {tags && (
        <SelectField
          value={tagSelected}
          label="Tag"
          options={tags.map(tag => ({
            displayed: `${tag.name} (${tag.amount})`,
            value: tag.name
          }))}
          setValue={setTag}
          resetValue={resetTag}
        />
      )}
    </>
  );
};

filters.propTypes = {
  showOptions: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  setShow: PropTypes.func.isRequired,
  show: PropTypes.string.isRequired,
  tagSelected: PropTypes.string,
  tags: PropTypes.array,
  setTag: PropTypes.func.isRequired,
  resetTag: PropTypes.func.isRequired,
  characNameSelected: PropTypes.string,
  characNames: PropTypes.array,
  setCharacName: PropTypes.func.isRequired,
  resetCharacName: PropTypes.func.isRequired
};

export default filters;
