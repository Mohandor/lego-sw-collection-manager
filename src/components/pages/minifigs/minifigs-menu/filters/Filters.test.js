import React from 'react';
import { shallow } from 'enzyme';
import Filters from './Filters';
import toJson from 'enzyme-to-json';

describe('<Filters />', () => {
  let props;
  beforeEach(() => {
    props = {
      showOptions: ['all', 'owned', 'missing'],
      show: 'all',
      setShow: jest.fn(),
      tagSelected: null,
      tags: [{ name: 'test', amount: 2 }],
      setTag: jest.fn(),
      characNameSelected: null,
      characNames: [{ name: 'test', amount: 2 }],
      setCharacName: jest.fn(),
    }
  });
  it('should render the Filters section', () => {
    const wrapper = shallow(<Filters {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});