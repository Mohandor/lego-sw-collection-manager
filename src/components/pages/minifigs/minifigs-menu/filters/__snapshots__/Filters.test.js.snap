// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> should render the Filters section 1`] = `
<Fragment>
  <inputs
    config={
      Object {
        "options": Array [
          "all",
          "owned",
          "missing",
        ],
        "row": true,
      }
    }
    label="Show"
    type="radiobuttons"
    updateInput={[MockFunction]}
    value="all"
  />
  <inputs
    className="padding10rem"
    config={
      Object {
        "options": Array [
          Object {
            "label": "test (2)",
            "value": "test",
          },
        ],
      }
    }
    label="Character Name"
    muiProps={
      Object {
        "InputLabelProps": Object {
          "shrink": true,
        },
        "variant": "outlined",
      }
    }
    placeholder="Filter by character name"
    type="autocomplete"
    updateInput={[MockFunction]}
    value={null}
  />
  <inputs
    config={
      Object {
        "options": Array [
          Object {
            "label": "test (2)",
            "value": "test",
          },
        ],
      }
    }
    label="Tag"
    muiProps={
      Object {
        "InputLabelProps": Object {
          "shrink": true,
        },
        "variant": "outlined",
      }
    }
    placeholder="Filter by tag"
    type="autocomplete"
    updateInput={[MockFunction]}
    value={null}
  />
</Fragment>
`;
